src/store/media-store.ts

```diff
@@ -222,11 +230,66 @@ export const createSettersSlice: StateCreator<
 
 	_handleProgress: (currentTime: number) =>
 		set(state => {
+			let shouldRunTimeAlarm = false;
 			const currentRelativeTime = Math.min(
 				state.endTime,
 				Math.max(0, currentTime - state.startTime),
 			);
+			let newTimeAlarm = {
+				current: state.currentTimeAlarm || 0,
+				next: state.nextTimeAlarm || 0,
+			};
+			const [timeAlarm, timeBeforeAlarm] = [
```

Comment:
Why do we do this? Can't we just access `state.timeAlarm` and `state.TimeBeforeAlarm` directly (consider that they are accessed only very seldom afterwards anyway)?